File: admin-fe/src/app/pages/workplace/model/workplace.ts
import {PlanInfo} from "../../payment-plan/model/plan-info";
import {AutocompleteItem} from "../../../../common/components/autocomplete/autocomplete-item";

export class Workplace implements AutocompleteItem {
  id: string | undefined;
  firstName: string | undefined;
  lastName: string | undefined;
  email: string | undefined;
  phone: string | undefined;
  plan: PlanInfo | undefined;
  status: string | undefined;
  createdDate: Date = new Date();
  getLabel(): string {
    return this.firstName || ' ' + ' ' + this.lastName || ' ';
  }
}

File: admin-be/src/main/java/com/emetaplus/admin/workplace/model/Workplace.java
package com.emetaplus.admin.workplace.model;

import com.emetaplus.admin.payments.model.Order;
import com.emetaplus.admin.plans.model.Plan;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.FieldNameConstants;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "workplaces")
@Getter
@Setter
@FieldNameConstants
public class Workplace {

    @Id
    @GeneratedValue
    private UUID id;
    private UUID workplaceId;
    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private LocalDateTime createdDate;
    private WorkplaceStatus status;
    private LocalDateTime expiredAt;

    private Integer freeDecksUsed = 0;
    private Integer freeBoardsUsed = 0;

    @ManyToOne
    private Plan plan;
    private LocalDateTime planPaidUntil;

    @OneToMany(mappedBy = "workplace")
    private List<Order> orders;
}
